src/StochasticNoiseSimulator.cpp

```diff
@@ -56,7 +58,7 @@ void StochasticNoiseSimulator::perfect_simulation_run() {
                     for (signed char i: quantum) {
```

Comment:
don't use `signed char` but rather use `dd::Qubit`. Please carefully check for such usages throughout this file.

Reply:
Done. I checked the other files as well